scripts/install-worker.sh

```diff
@@ -111,6 +115,19 @@ sudo systemctl restart sshd.service
 sudo mkdir -p /etc/eks
 sudo mv $TEMPLATE_DIR/iptables-restore.service /etc/eks/iptables-restore.service
 
+################################################################################
+### installing updated cli #####################################################
```

Comment:
I think we should do this in a separate PR. What's the dependency on it here?

Reply:
`describe-add-on versions` is not in the aws-cli shipped currently

Reply:
Just pop up in my head... a _**nice to have**_ enhancement,

Previously, it only require EC2 level permission, but now it required `eks:DescribeAddonVersions`.
Should it be mentioned in somewhere for extra permission?

ref: https://docs.aws.amazon.com/eks/latest/APIReference/API_DescribeAddonVersions.html

Reply:
We should mention this in our "Building the AMI" docs, yeah.

Reply:
i think we should move this: https://github.com/awslabs/amazon-eks-ami/pull/938/files#diff-dff0aa3a99c138c812c9364e64a6f95abec4590a27f0fb2b448231342b4a3b89R70

Into an `else` here. If we can install the updated CLI bundle, we can avoid installing all the global python dependencies for the yum package, e.g.:
```
> yum deplist awscli
                                                  
package: awscli.noarch 1.18.147-1.amzn2.0.1
  dependency: /usr/bin/python2
   provider: python.x86_64 2.7.18-1.amzn2.0.5
  dependency: PyYAML
   provider: PyYAML.x86_64 3.10-11.amzn2.0.2
  dependency: groff-base
   provider: groff-base.x86_64 1.22.2-8.amzn2.0.2
  dependency: python(abi) = 2.7
   provider: python.x86_64 2.7.18-1.amzn2.0.5
  dependency: python-colorama
   provider: python-colorama.noarch 0.3.2-3.amzn2
  dependency: python-docutils
   provider: python-docutils.noarch 0.12-0.2.20140510svn7747.amzn2
  dependency: python-rsa
   provider: python2-rsa.noarch 3.4.1-1.amzn2.0.1
  dependency: python2-botocore = 1.18.6
   provider: python2-botocore.noarch 1.18.6-1.amzn2.0.1
  dependency: python2-s3transfer
   provider: python2-s3transfer.noarch 0.3.3-1.amzn2.0.1
```

Reply:
moved awscli install to install only if the awscli v2 can't be installed.

Reply:
I've added docs on the image caching including permissions for the AMI.